### ISSUE TITLE
Message for Custom Rule

### ISSUE THREAD

**Issue by jylinman** [tgriesser/checkit#45]
Hey Guys,

I am trying to add custom error messages for custom rules that I add into the CheckIt object.  For instance, this works fine:

```
var rules = {
  title: [{
    rule: 'required',
    message: 'Required'
  }]
};
```
That returns the expected `Required` message.  However, if I want to add a unique function and change it's message, it returns null:

```
var rules = {
  title: [{
        rule: function(val) {
          return new Promise(function(resolve, reject) {
            User.where('email', '=', val).count('id').then(function(res) {
              if (res > 0) return reject(true);

              resolve(true);
            });
          });
        },
        message: 'Exists'
      }]
};
```
The function runs properly, but the message is `null`.  Any thoughts on customizing the message for custom rules?

**Comment by rhys-vdw** [comment]
Hi @jylinman, I'm helping to maintain this project because it relates to Bookshelf, but have little knowledge of its API and inner workings.

I'll have to leave it to you (or another user) to answer/investigate this one. If you do find the answer, or would like a change, I'll happily merge a PR to update behaviour or clarify docs, but I'm no better equipped to answer specific questions than you are! That said this sounds like an important feature so I'd appreciate if you do report back. Cheers!

**Comment by jylinman** [comment]
Hi @rhys-vdw thanks for your fast reply-- I used bookshelf in this example, but don't necessarily need a bookshelf specific example...  It can be any type of custom rule, which would make it framework agnostic, but I am more curious about the underlying issue of the two examples but one example returns the custom message and the other doesn't.  I may have time to look into the issue, but am curious if you knew where this could be...

**Comment by rhys-vdw** [comment]
> but am curious if you knew where this could be...

Not really (sorry), but the code is pretty straight forward if I recall.

**Comment by jylinman** [comment]
I phrased the question incorrectly-- I am trying to say that this is a bug.  If I provide a function instead of a string, the message doesn't get returned.

**Comment by rhys-vdw** [comment]
Oh, sorry @jylinman, I hadn't understood. Yep, agree that this is a bug - thanks for the issue. I'll accept a PR fixing it. :+1:

**Comment by cecchi** [comment]
Agreed this is a bug, but for those looking for a workaround you can throw an error with a message from the custom validation function:

```js
const rules = {
    'password' : [
        {
            'rule'    : 'required',
            'message' : 'Password is required.'
        },
        {
            'rule'    : 'minLength:8',
            'message' : 'Password must be at least 8 characters.'
        },
        function(val) {
            if(!/[0-9]/.test(val)) {
                throw new Error('Password must contain a number.');
            }
        }
    ]
}
```
